providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/connector/grpc_watcher.py

```diff
@@ -60,7 +61,17 @@ def _generate_channel(self, config: Config) -> grpc.Channel:
             ("grpc.max_reconnect_backoff_ms", config.retry_backoff_max_ms),
             ("grpc.min_reconnect_backoff_ms", config.stream_deadline_ms),
         ]
-        if config.tls:
+        if config.default_authority is not None:
```

Comment:
```suggestion
        if config.default_authority:
```
this makes it a truthy check, which is a more Pythonic approach

Reply:
But empty is falsy, right? I guess it might be weird to have an empty authority but if the user tried to set empty explcitly, shouldn't we allow it?

Reply:
@aepfli do you know by any chance if those values can be empty? I couldn't find much information on validation for it.

Reply:
good question to be fair - i never checked how we treat empty strings for eg. selector, not sure how flagd is reacting if it is empty (assumption would be, that it is treated that same way) - might be worth to adapt our gherkin files to also cover such cases ;)